providers/flagd/pkg/service/service.go

```diff
@@ -190,21 +197,46 @@ func handleError(err error) openfeature.ResolutionError {
 }
 
 // EventStream emits received events on the given channel
-func (s *Service) EventStream(ctx context.Context, eventChan chan<- *schemaV1.EventStreamResponse, errorChan chan<- error) {
+func (s *Service) EventStream(
+	ctx context.Context, eventChan chan<- *schemaV1.EventStreamResponse, maxAttempts int, errChan chan<- error,
+) {
+	delay := s.baseRetryDelay
+	var err error
+	for i := 1; i <= maxAttempts; i++ {
+		log.Infof("attempt %d at connecting to event stream", i)
+		i, err = s.eventStream(ctx, eventChan, i, maxAttempts)
+		if i == 1 {
+			delay = s.baseRetryDelay // reset delay if the connection was successful before failing
+		}
+		if err != nil && i <= maxAttempts {
+			delay = 2 * delay
+			log.Infof("connection to event stream failed, sleeping %v", delay)
+			time.Sleep(delay)
+		}
+	}
+
+	if err != nil {
+		errChan <- err
+	}
+}
```

Comment:
I wonder if messages related to reconnect and giving up connection should be `warning` level.